install-dependencies.sh

```diff
@@ -107,6 +110,7 @@ fedora_packages=(
     rust-std-static-wasm32-wasi
     wabt
     binaryen
+    lcov
```

Comment:
This has (at least on my Fedora) hefty deps:
```
Install  34 Packages

Total download size: 6.3 M
Installed size: 35 M
```

Why would we run it on all deployments, instead of when running code coverage?

Reply:
To make a long story short, test.py (which is run from inside dbuild) will process coverage data, this in turn will require lcov.
I could split it but then after running test.py it will require manual steps to produce final report, the end result should be that you run test.py and you get a report which you can use for example to prove that your code is tested.

Reply:
> This has (at least on my Fedora) hefty deps:
> 
> ```
> Install  34 Packages
> 
> Total download size: 6.3 M
> Installed size: 35 M
> ```
> 
> Why would we run it on all deployments, instead of when running code coverage?

In the dbuild system itself, it looks like it only bring is some pearl deps:
```
Installing dependencies:
 perl-GD ...                 fedora                                                                   180 k
 perl-PerlIO-gzip ...   fedora                                                                    26 k

Install  3 Packages

Total download size: 306 k
Installed size: 825 k
Is this ok [y/N]: 
```

Also it is untrue that we run it on deployments, only in dbuild. It is not going to be part of the packaging since none of the scylla binaries nor the python packages that we distribute depends on it.

Reply:
Interesting, are you saying that the dbuild image *already* had Perl itself? Do you know why?
Don't get me wrong, I love Perl and 20 years ago, it was my favorite programming language. But I didn't realize we still use it for anything.

Reply:
> Interesting, are you saying that the dbuild image _already_ had Perl itself? Do you know why?

```

Reply:
> Interesting, are you saying that the dbuild image _already_ had Perl itself? Do you know why?

```
$ tools/toolchain/dbuild sudo repoquery --userinstalled --whatrequires perl-interpreter
debhelper-0:13.11.6-1.fc38.noarch
devscripts-0:2.23.0-1.fc38.x86_64
doxygen-2:1.9.6-7.fc38.x86_64
git-0:2.42.0-2.fc38.x86_64
stow-0:2.3.1-10.fc38.noarch
```

Reply:
Actually, there's more:
```
$ tools/toolchain/dbuild sudo repoquery --userinstalled --recursive --whatrequires perl-interpreter
debhelper-0:13.11.6-1.fc38.noarch
devscripts-0:2.23.0-1.fc38.x86_64
doxygen-2:1.9.6-7.fc38.x86_64
git-0:2.42.0-2.fc38.x86_64
hwloc-devel-0:2.9.3-1.fc38.x86_64
libtool-ltdl-devel-0:2.4.7-6.fc38.x86_64
stow-0:2.3.1-10.fc38.noarch
valgrind-devel-1:3.21.0-9.fc38.x86_64
```

Reply:
Weird.... Why would hwloc, git, and all that other stuff, require Perl? But who am I to argue with output of commands :-)
Anyway, since it does, I think all this discussion is based on a mistake. @mykaul are you sure lcov added 34 packages for you? Do you know which?

Reply:
> Weird.... Why would hwloc, git, and all that other stuff, require Perl?

Why is it weird? It is (or was) a common script language, many things depend on it. There are 30k lines of perl in git's repository.

Reply:
You're right, I checked and the `git` package in Fedora has 6 Perl scripts. We probably don't need any of them for Scylla builds, but what the heck.

Reply:
Incidentally, `dbuild` currently contains 226 perl libraries:
```
$ tools/toolchain/dbuild sudo repoquery --installed | grep ^perl- | wc -l
226
```
This PR adds another two.

Reply:
> nyway, since it does, I think all this discussion is based on a mistake. @mykaul are you sure lcov added 34 packages for you? Do you know which?

I believe that @mykaul tried it on his own system and not from within dbuild, probably `install-dependencies.sh` prior to my change was never run on this system.